Iterate on design & add PWA support

brettchalupa opened the pull request:
This iterates on the UI to make it a bit more user friendly in a variety of ways. I focused on keeping the UI simple, accessible, and semantic.

I specifically didn't change any JS and tried to not change anything relating to the core functionality. It's mostly CSS and some HTML adjustments.

See commits for details, I tried to make them small and atomic.

In addition to using some brand colors and styles, I've also added the logo, expanded on the README, and made it so that the website can be installed as a Progressive Web App to make it possible for Chrome users to install on their desktop for easy access.

## Screenie

![screenshot of the LED Matrix Input Module control](https://github.com/FrameworkComputer/dotmatrixtool/assets/928367/bd8d3986-aa42-4d79-9d6e-4ef77945b7d9)

### Progressive Web App Install UI

![image](https://github.com/FrameworkComputer/dotmatrixtool/assets/928367/95b02f57-857f-4d4b-b320-6a98c5bc7621)


## Acceptance Criteria

1. Check out this PR - `gh pr checkout 6`
2. Boot up a web server - if you have Python, `python3 -m http.server` or with Ruby: `ruby -run -e httpd . -p 5555`
3. View the site locally and ensure it:
   1. looks nice
   2. works with input modules if ya' got em

Happy to iterate on this and all feedback welcome.

brettchalupa committed:
chore: add Framework to LICENSE copyright

brettchalupa committed:
chore: revise README

expands on it a little bit and adds some more info

brettchalupa committed:
chore: refine page metadata desc

brettchalupa committed:
feat: add framework cog favicon

brettchalupa committed:
feat: more specific page title

brettchalupa committed:
chore: improve text constrast

this makes it so that the text on the page and the background have a
higher constrast so that it's more accessible. it also begins to
introduce some CSS variables that'll be used for color as the styles are
iterated on

brettchalupa committed:
fix: semantic h1

brettchalupa committed:
chore: sans-serif font

more "on brand"

intentionally just using helvetica or system sans-serif so the tool can
work offline and look decent

brettchalupa committed:
chore: revise color vars to match brand

makes the dark color the same as the frame.work black and fixes color
issue with select element text

brettchalupa committed:
chore: semantic headers & layout & spacing revisions

just making some passes on improving it

brettchalupa committed:
feat: promote footer into left sidebar section

raises its visibility and has a better layout/flow than the way the
footer was

brettchalupa committed:
chore: revise light color

makes it a bit dimmer so people don't have their eyes recoil due to too much contrast

brettchalupa committed:
feat: display the framework logo

~so on brand~

brettchalupa committed:
chore: improve layout on large displays

centers content and makes the left sidebar not be super wide; overall
looking nice on a variety of monitor sizes

brettchalupa committed:
feat: installable Progressive Web App

this makes it so people can install it as a progressive web app on
from Chrome, treating it like a desktop app

More details: https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Guides/Making_PWAs_installable

brettchalupa committed:
chore: make Input Modules proper noun in README

checked in and this is correct

brettchalupa committed:
fix: make alt text for logo accurate

brettchalupa committed:
chore: revise instructions text

brettchalupa committed:
fix(docs): README typo

brettchalupa committed:
chore: add HTML validator & GH workflow

to ensure the HTML is valid which is important for accessibility reasons
and making it easier to contribute

brettchalupa committed:
fix: make html valid

makes use of the new `make validate` command that checks the
`index.html` file; which I used to determine issues to fix. many of them
were related to inline styles, ids starting with `_`, and some link
element locations and script props.

brettchalupa committed:
chore: add copyright year to LICENSE

it was missing and should be there

zachfeldman commented on README.md:
```diff
@@ -1,24 +1,33 @@
 
-Click and drag to draw, CTRL+click to erase.
+Use the "Connect Left" and "Connect Right" buttons to connect to each
+respective input module. Draw on each input browser in your browser, and it
```
Nit: input browser -> input module?

zachfeldman commented on README.md:
```diff
@@ -1,24 +1,33 @@
+
+## Contributing
+
+Contrirbutions are welcome. Submit pull requests to the `usbserial` branch.
```
Contrirbutions -> Contributions

zachfeldman replied:
Whoops you already fixed this!

zachfeldman commented on index.html:
```diff
@@ -6,7 +6,7 @@
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags -->
-    <meta name="description" content="An online tool for controlling the Framework LED Matrix">
+    <meta name="description" content="An web tool for controlling the Framework LED Matrix input module">
```
An -> A

zachfeldman replied:
Also maybe we want to add "for the Framework Laptop 16" to the end?

zachfeldman commented on manifest.json:
```diff
@@ -0,0 +1,12 @@
+{
```
Very cool idea, love it! FYI it only worked when running it with the Ruby server. It'll probably work better on https://ledmatrix.frame.work/.

brettchalupa replied:
Yeah, I'd assume that's a limitation of how PWAs work. If someone downloads the files and uses the download, I think it's safe to assume they don't need the PWA installed then.

zachfeldman commented on README.md:
```diff
@@ -31,3 +31,8 @@ See also http://dotmatrixtool.com
+
+When preparing your pull request, please be sure:
+
+1. the tool still works with connected LED Matrix Input Modules
```
Nit: the -> The

brettchalupa replied:
I prefer it being lower case since they're not proper sentences

zachfeldman commented on app.js:
```diff
@@ -116,8 +116,8 @@ function updateTableLeft() {
   $table_left = populateTable(null, height, width, "left");
-	$('#_grid_left').html('');
-	$('#_grid_left').append($table_left);
+	$('#led-grid_left').html('');
```
Nice and clean :shower:

zachfeldman reviewed (approved):
Thanks @brettchalupa , this is a huge improvement!

It is working well locally with my LED matrix. Only issue is that after some time I saw this JS error, even when refreshing the page. May need some help from @JohnAZoidberg to fix:
![image](https://github.com/FrameworkComputer/dotmatrixtool/assets/1437644/e4a626c7-367c-4cc9-99b2-b070bb61a19d)
it did work again when I tried it with the PWA! So I bet it would work again if I restarted Chrome.

We may want to give credit to the original person that created this tool on the page, I think that would be cool - see the README "This repository is based on code from https://github.com/stefangordon/dotmatrixtool."

brettchalupa committed:
chore(docs): mention where the site is hosted

just to make it clear how it works for future ppl

brettchalupa committed:
fix(docs): typo about usage in README

brettchalupa committed:
fix: instructions text for Connect button

brettchalupa committed:
fix: meta desc to be specific & on brand

brettchalupa committed:
chore: thank original author

brettchalupa committed:
chore: bring back monospaced font

hacker vibes

suggested by @bonkrat

brettchalupa commented:
new screenie w/ hacker vibes bringing back the monospace font

<img width="682" alt="image" src="https://github.com/FrameworkComputer/dotmatrixtool/assets/928367/3a365c2a-e1f3-4222-906c-af14b317d7cb">

brettchalupa commented:
@zachfeldman thanks for all of the feedback, I believe I've addressed most of the issue and implemented the suggestions. 🙌 

Could you log a separate GitHub issue in this repo for the console error? That seems unrelated to the changes and isn't something I can fix yet (you or @JohnAZoidberg could investigate though if you're feeling interested 😸 ).

brettchalupa commented:
I'm going to rebase this on top of the latest in `usbserial` so that it gets the Cloudflare Pages preview deployment for easy review + fixes some conflicts I introduced.

brettchalupa commented:
yay the preview deploys are working now, that's legit! you can see them as part of the PR's deployments if you want to give it a test on a hosted version

brettchalupa committed:
chore: switch font to Noto Sans Mono

feedback from anne! https://fonts.google.com/noto/specimen/Noto+Sans+Mono

brettchalupa committed:
fix: responsive columnar layout on lil screens

just in case ppl view on mobile and see a slightly better experience

brettchalupa committed:
fix: better align the brightness slider

it was just a little misaligned with the text and the slider + label
could use a little more spacing

brettchalupa committed:
chore: make brightness slider label a label

for semantic correctness and more accessible

brettchalupa committed:
chore: make left & right label bold

makes it distinguished which I think is more clear

zachfeldman reviewed (approved):
Thanks for addressing all of my comments @brettchalupa - still working great with my LED Matrices!
